### ISSUE TITLE
BlobHash reverts in cast, but succesful on-chain

### ISSUE THREAD

**Issue by Michaelr-spherex** [foundry-rs/foundry#7426]
### Component

Cast

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (42a9d34 2024-03-18T00:19:07.645071938Z)

### What command(s) is the bug in?

cast run --rpc-url https://rpc.sendblocks.io/eth/0gZiGozbvL7uTKFRYeMu719RNbz6e7e12QpkVCuW 0x6118e4e343ab6eb261ca40d00c8afcc5523045acfc4043c5d1a71de5f35b055a

### Operating System

Linux

### Describe the bug

regarding OS: I run it in windows WSL, so, it's supposed linux.

The short version is the provided command runs a tx that REVERTS, but the original tx ([here](https://etherscan.io/tx/0x6118e4e343ab6eb261ca40d00c8afcc5523045acfc4043c5d1a71de5f35b055a)) is succesful.

It seems that the problem is with the BLOBHASH opCode, because after it, the tx reverts. But maybe I understand the debug data wrong and it's a different reason.

**Comment by mattsse** [comment]
ref https://github.com/foundry-rs/compilers/pull/94

we're not detecting the proper evm version here, until fixed properly you can run with
--evm-version
